Prog/Calculate_equity_gaps.do

```diff
@@ -0,0 +1,57 @@
+* Program: Calculate_equity_gaps
+* Created by: Yipeng Su
+* Last modified: 10/22/2018
+
+import excel using "L:\Libraries\Equity\Prog\JPMC feature\For stata.xlsx", firstrow
+
+drop GID
+
+*index is a simplified ID variable I created for each of the equity variables so we can reshape
+reshape wide numerator denom equityvariable, i(index) j(geo) string
+
+save "L:\Libraries\Equity\Prog\JPMC feature\Wide data.dta", replace
+
+*merge the wide file back to long file for calculating gaps
+import excel using "L:\Libraries\Equity\Prog\JPMC feature\For stata.xlsx", firstrow clear
+
+merge m:1 index using "L:\Libraries\Equity\Prog\JPMC feature\Wide data.dta"
+
+*calculate equity gap = (dealta equity var)*denominator of each geography of interest
+local geography DC Ward1 Ward2 Ward3 Ward4 Ward5 Ward6 Ward7 Ward8 Cluster1 Cluster2 Cluster3 Cluster4 Cluster5 Cluster6 ///
+                Cluster7 Cluster8 Cluster9 Cluster10 Cluster11 Cluster12 Cluster13 Cluster14 Cluster15 Cluster16 Cluster17 ///
+				Cluster18 Cluster19 Cluster20 Cluster21	Cluster22 Cluster23 Cluster24 Cluster25 Cluster26 Cluster27 Cluster28 Cluster29 ///
+				Cluster30 Cluster31 Cluster32 Cluster33 Cluster34 Cluster35 Cluster36 Cluster37 Cluster38 Cluster39 Cluster40 Cluster41 Cluster43 Cluster44
+				
+forvalues i= 1/22 { 
+  
+	foreach geo of local geography {
+	gen `geo'_index`i'=.
+	replace `geo'_index`i' = (equityvariable-equityvariable`geo')*denom
+
+	}
+
+}
+
+sort index GID
+
+
+*export the square shaped dataset
+levelsof index, local(indicatorvar)
+foreach var of local indicatorvar {
+    preserve
+    export excel indicator geo GID DC_index`var' Ward1_index`var' Ward2_index`var' Ward3_index`var' Ward4_index`var' Ward5_index`var' ///
```

Comment:
could you add the equityvariable rate for the row geography to the output for reference?